packages/headless/src/features/insight-search/insight-search-analytics-actions.ts

```diff
@@ -8,7 +8,12 @@ import {getQueryInitialState} from '../query/query-state';
 export const logFetchMoreResults = makeInsightAnalyticsAction(
   'search/logFetchMoreResults',
   AnalyticsType.Search,
-  (client) => client.logFetchMoreResults()
+  (client, state) =>
+    client.logFetchMoreResults({
+      caseContext: state.insightCaseContext?.caseContext || {},
```

Comment:
May be create an exported  function that state as param and return this object, to be used in all project, may be one day we need to add something , it would be easier to make change
what do you think?

Reply:
I absolutely agree! Good idea 👌